### PR TITLE
Create VeteranRecordRequest for non-comp request issues

### PR DESCRIPTION
**Pull request by pkarman**
connects #8674 

### Description
A new task `VeteranRecordRequest` created whenever an Appeal request issue has a benefit_type that is not compensation or pension.

### CODE REVIEW

File: app/controllers/decision_reviews_controller.rb [hunk 1]
```diff
@@ -40,15 +40,19 @@ def task_id
   end
 
   def task
-    @task ||= DecisionReviewTask.find(task_id)
+    @task ||= Task.includes([:appeal, :assigned_to]).find(task_id)
```
Comment (pkarman) [inline app/controllers/decision_reviews_controller.rb]: we have more than one kind of task subclass now.

File: app/controllers/decision_reviews_controller.rb [hunk 2]
```diff
@@ -40,15 +40,19 @@ def task_id
...
   end
 
   def in_progress_tasks
-    apply_task_serializer(business_line.tasks.order(assigned_at: :desc).reject(&:completed?))
+    apply_task_serializer(
+      business_line.tasks.includes([:assigned_to, :appeal]).order(assigned_at: :desc).reject(&:completed?)
```
Comment (pkarman) [inline app/controllers/decision_reviews_controller.rb]: this just quiets the bullet warnings about eager loading.

File: app/models/appeal.rb [hunk 3]
```diff
@@ -329,6 +330,18 @@ def program
 
   private
 
+  def create_business_line_tasks
+    request_issues.select(&:requires_record_request_task?).each do |req_issue|
+      business_line = req_issue.business_line
+      VeteranRecordRequest.create!(
+        parent: root_task,
```
Comment (pkarman) [inline app/models/appeal.rb]: not sure if we need to make this the child of the Root task or not.

File: app/models/serializers/work_queue/decision_review_task_serializer.rb [hunk 4]
```diff
@@ -17,7 +17,7 @@ def claimant_name
   end
 
   def claimant_relationship
-    return "self" unless decision_review.claimants.any?
+    return "self" unless decision_review.veteran_is_not_claimant
```
Comment (pkarman) [inline app/models/serializers/work_queue/decision_review_task_serializer.rb]: fixes logic that would bite us eventually.

**Comment by pkarman** [comment]
screen shot 2019-01-29 at 3 43 17 pm

File: app/models/serializers/work_queue/decision_review_task_serializer.rb (hunk 4 above)
Comment (leikkisa) [reply app/models/serializers/work_queue/decision_review_task_serializer.rb]: 👍

File: app/models/appeal.rb (hunk 3 above)
Comment (pkarman) [reply app/models/appeal.rb]: confirmed yes with @lowellrex

**Pull request merged**